packages/commands/src/track/upload-track.ts

```diff
@@ -0,0 +1,271 @@
+    'Track preview start time (seconds)',
+    parseInt
+  )
+  .option('-l, --license <license>', 'License of track')
+  .option('-f, --from <from>', 'The account to upload track from')
+  .option(
+    '-u, --price <price>',
+    'Generate a stream conditions object with the given price in cents. Cannot be used with -r'
+  )
+  .option(
+    '-r, --stream-conditions <stream conditions>',
+    'Manually set a stream conditions object. Cannot be used with -u',
+    ''
+  )
+  .option(
+    '-o, --is-downloadable <is downloadable>',
+    'Whether track is downloadable'
+  )
+  .option(
+    '-dp, --download-price <download price>',
+    'Generate a download conditions object with the given price in cents. Cannot be used with -dc'
+  )
+  .option(
+    '-dc, --download-conditions <download conditions>',
+    'Manually set a download conditions object. Cannot be used with -dp',
+    ''
+  )
+  .option(
+    '--remix-of <remix-of>',
+    'Specify the track ID of the original track if this is a remix'
+  )
+  .option(
+    '--output <path>',
+    'A path to which to write a json file containing the track data'
+  )
+  .option(
+    '-h, --hidden',
+    'Whether the track is hidden from the feed (ie. is unlisted)'
+  )
+  .action(async (track, { title = `Track ${randomBytes(2)
+      .toString('hex')
+      .padStart(4, '0')
+      .toUpperCase()}`, tags, description = 'Created with audius-cmd', mood = randomMood(), genre = randomGenre(), previewStartSeconds, license, from, price, streamConditions, isDownloadable, downloadPrice, downloadConditions, remixOf, output, hidden }) => {
+    const audiusSdk = await initializeAudiusSdk({ handle: from })
+    const userId = await getCurrentUserId()
+
+    let trackStream
+    if (track.startsWith('%')) {
+      // %filesize
+      const match = track.match(/%(\d+)(.*)/)!
+      let size = parseInt(match[1])
+      let unit = match[2]
+      if (unit === 'm') {
+        size *= 1024 * 1024
+      } else if (unit === 'k') {
+        size *= 1024
+      } else {
+        throw new Error(`Unknown unit "${unit}"`)
+      }
+
+      await generateWhiteNoise(size / 8064, '/tmp/audius-cmd.mp3')
+      trackStream = createReadStream('/tmp/audius-cmd.mp3')
+    } else if (track.startsWith(':')) {
+      // :/path/to/track
+      trackStream = createReadStream(track.slice(1))
+    } else {
+      throw new Error(`Failed to parse track "${track}"`)
+    }
+
+    const parsedStreamConditions = await getStreamConditions({
+      streamConditions,
+      price,
+      audiusSdk
+    })
+    const parsedDownloadConditions = await getDownloadConditions({
+      streamConditions,
+      parsedStreamConditions,
+      downloadConditions,
+      downloadPrice,
+      audiusSdk
+    })
+
+    const { trackId } = await audiusSdk.tracks.uploadTrack({
+      userId,
+      coverArtFile: { buffer: createRandomImage(), name: 'coverart' },
+      trackFile: { buffer: await readStreamToBuffer(trackStream) },
```

Comment:
we lost support for passing in the ReadStream directly unfortunately :(

Reply:
we could probably add it back if we want!